Update documentation

pezholio opened the pull request:
This updates the documentation to make the project easier to get to grips with, as well as using the scripts to rule them all pattern across the project and updating some of the CI

pezholio committed:
Use ruby/setup-ruby

This does the caching automatically, as well as
runs bundler

dragon-dxw commented on README.md:
```diff
@@ -2,16 +2,40 @@
 # Golden Retriever
 
-Imports opportuntities from the [Digital Marketplace](https://www.digitalmarketplace.service.gov.uk/digital-outcomes-and-specialists/opportunities) into Hubspot
+Golden Retriever is imports opportuntities from the [Digital Marketplace](https://www.digitalmarketplace.service.gov.uk/digital-outcomes-and-specialists/opportunities) into Hubspot and drops an alert into our Slack like this:
```
Delete 'is': "Golden Retriever imports opportunities"

mec commented on README.md:
```diff
@@ -2,16 +2,40 @@
+
+## Environment Variables
+
+For the app to run properly, you need a to have the following environment variables
```
Does the app work at all without these? If not I would be templeted to say these are required, something like:

> You must have the following required environment variables set in hte .env file for the application to function as expected:

Might also be worth mentioning that the setup script asks for them?

mec commented on .github/workflows/build.yml:
```diff
@@ -11,24 +11,15 @@ on: [push]
-        bundle install --jobs 4 --retry 3 --path ./vendor/bundle
-        bundle exec rake
+      - uses: actions/checkout@v1
+      - uses: ruby/setup-ruby@v1
```
👌

mec commented on script/cibuild:
```diff
@@ -0,0 +1,11 @@
+
+cd "$(dirname "$0")/.."
+
+bundle exec rspec
```
Why did the end of this change from `bundle exec rake` to `bundle exec rspec` is `rspec` just the default? I am being lazy and could probably look…

pezholio replied:
No reason really. I might as well make it consistent though 👍

mec replied:
I definately prefer to be explicit and avoid the default task and the naked `rake` expecially in scripts? 👌

mec reviewed (changes requested):
Looks good, one question on the `cibuild` script for my own learning, answer that and I'll give you the ship. :)

pezholio committed:
Update README

This gives wider context on how to use and setup
the application, as well as adding a setup script to
get the application into a running state.

pezholio committed:
Add more scripts to rule them all

We're using these to set up, so it makes sense to
use these elsewhere
